tests/Backend/ExternalBuckets/SnowflakeRegisterBucketTest.php

```diff
@@ -656,6 +656,118 @@ public function testRegistrationOfView(): void
         $this->assertEquals('view', $view['tableType']);
     }
 
+    public function testAlteredColumnThrowsUserExAndAfterRefreshWillWork(): void
+    {
+        $description = $this->generateDescriptionForTestObject();
+        $testBucketName = $this->getTestBucketName($description);
+        $bucketId = self::STAGE_IN . '.' . $testBucketName;
+
+        $this->dropBucketIfExists($this->_client, $bucketId, true);
+
+        $ws = new Workspaces($this->_client);
+        $workspace = $ws->createWorkspace();
+
+        $db = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $db->createTable('MY_LITTLE_TABLE_FOR_VIEW', ['AMOUNT' => 'NUMBER', 'DESCRIPTION' => 'TEXT']);
+
+        $db->executeQuery(
+            <<<SQL
+CREATE OR REPLACE VIEW MY_LITTLE_VIEW AS SELECT * FROM  MY_LITTLE_TABLE_FOR_VIEW;
+SQL,
+        );
+
+        $idOfBucket = $this->_client->registerBucket(
+            $testBucketName,
+            [$workspace['connection']['database'], $workspace['connection']['schema']],
+            'in',
+            'Iam in workspace',
+            'snowflake',
+            'Iam-your-external-bucket-to-test-alter-table',
+        );
+
+        $bucket = $this->_client->getBucket($idOfBucket);
+        $this->assertTrue($bucket['hasExternalSchema']);
+        $this->assertSame($workspace['connection']['database'], $bucket['databaseName']);
+
+        // check table existence
+        $tables = $this->_client->listTables($idOfBucket);
+        $this->assertCount(2, $tables);
+        $table = $tables[0];
+        $view = $tables[1];
+
+        // now add new column to table and recreate view, to test preview still works
+        $db->executeQuery(
+            <<<SQL
+ALTER TABLE MY_LITTLE_TABLE_FOR_VIEW ADD COLUMN AGE INTEGER;
+SQL,
+        );
+
+        $db->executeQuery(
+            <<<SQL
+CREATE OR REPLACE VIEW MY_LITTLE_VIEW AS SELECT * FROM  MY_LITTLE_TABLE_FOR_VIEW;
+SQL,
+        );
+
+        /** @var array $tableDataPreview */
+        $tableDataPreview = $this->_client->getTableDataPreview($table['id'], ['format' => 'json']);
```

Comment:
preview "vracia" string aspon podla doc, nechem teraz vrtat do toho docu pretoze to rozbilo zas veci niekde inde